test/test_web.py

```diff
@@ -4693,6 +4693,92 @@ def test_48_task_presenter_editor_works_json(self, mock):
         assert data['form']['guidelines'] == 'Some guidelines!', data
 
 
+    @with_context
+    @patch('pybossa.view.projects.uploader.upload_file', return_value=True)
+    def test_task_presenter_large_image_upload(self, mock):
+        """Test API /tasks/taskpresenterimageupload should not upload images with size > 5 MB"""
+        print("running test_task_presenter_image_upload...")
+        user = UserFactory.create(id=500)
+        project = ProjectFactory.create(
+            short_name='test_project',
+            name='Test Project',
+            info={
+                'total': 150,
+                'task_presenter': 'foo',
+                'data_classification': dict(input_data="L4 - public", output_data="L4 - public"),
+                'kpi': 0.5,
+                'product': 'abc',
+                'subproduct': 'def',
+            },
+            owner=user)
+        headers = [('Authorization', user.api_key)]
+        with open('./test/files/large-image.jpg', 'rb') as img:
+            imgStringIO = BytesIO(img.read())
+        # Call API method to upload image.
+        res = self.app.post('/project/{}/tasks/taskpresenterimageupload'.format(project.short_name), headers=headers, data={'image': (imgStringIO, 'large-image.jpg')})
+        res_data = json.loads(res.data)
+        assert res.status_code == 200, "POST image upload should be successful"
```

Comment:
Consider adding a check that `res_data['errors'] == True`. Additionally, if the HTTP status code is changed to a 500, we should check for that value here rather than 200.